source/common/ssl/certificate_validation_context_config_impl.cc

```diff
@@ -51,5 +53,42 @@ CertificateValidationContextConfigImpl::CertificateValidationContextConfigImpl(
   }
 }
 
+std::vector<envoy::extensions::transport_sockets::tls::v3::SubjectAltNameMatcher>
+CertificateValidationContextConfigImpl::getSubjectAltNameMatchers(
+    const envoy::extensions::transport_sockets::tls::v3::CertificateValidationContext& config) {
+  if (!config.match_typed_subject_alt_names().empty() &&
+      !config.match_subject_alt_names().empty()) {
+    throw EnvoyException("SAN-based verification using both match_typed_subject_alt_names and "
+                         "the deprecated match_subject_alt_names is not allowed");
+  }
+  std::vector<envoy::extensions::transport_sockets::tls::v3::SubjectAltNameMatcher>
+      subject_alt_name_matchers(config.match_typed_subject_alt_names().begin(),
+                                config.match_typed_subject_alt_names().end());
+  // Handle deprecated string type san matchers without san type specified, by
+  // creating a matcher for each supported type.
+  for (auto& matcher : config.match_subject_alt_names()) {
+    subject_alt_name_matchers.emplace_back();
```

Comment:
You may be able to iterate over the values [`SanType_MIN`, `SanType_MAX`] (see https://developers.google.com/protocol-buffers/docs/reference/cpp-generated#enum).

Reply:
I'd rather be explicit here. If new enum types are added in the future, those should NOT be iterated over in this loop, as this is for backwards compatibility ONLY.

Reply:
So in this case I suggest to refactor the code to iterate over the specific set of enum values, and add them (decrease code duplication).

Reply:
Fixed